### PR TITLE
refactor(routable/VItemGroup): normalize behavior

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Normalized functionality of **routable** and **item-group** items to reduce duplicated code and differing functionality.

## Motivation and Context
Controlling the color of items that were active (through groups and routes) was different depending upon which component you were using. It becomes easier to maintain these items when they all have the same expected behavior. Color is now inherited from the parent in the context of groups and can be overwritten by usage of an `active-class`.

### BREAKING CHANGES:

- `v-bottom-navigation`
  - **color** -> **background-color**
  - **color** now affects active `v-btn` color
- `v-chip`
  - removed **selected** prop
  - use **input-value**
- `v-list-item-group`
  - removed default **active-class**, must provide **color**

### FEATURES
- `v-chip`
  - now implements the **routable** mixin

### MAJOR CHANGES
- **active-class** is no longer used to specify color of children
  - use **color**
  - default item active classes are now set using **proxyClass**, similar to `v-list-item`
- **routable**
  - mixin now infers **classes** and **styles** locally
  - moved in duplicated functionality from multiple implementing components

## How Has This Been Tested?
jest / visual

## Markup:

<details>

```vue
<template>
  <v-card
    max-width="400"
    class="mx-auto"
  >
    <v-card-text>
      <v-chip>Hello</v-chip>
      <v-chip
        color="purple"
        dark
        to="/page2"
      >Hello</v-chip>
      <div class="my-3"></div>
      <v-btn to="/page1">Button</v-btn>
      <v-btn to="/page2">Button 2</v-btn>
      <div class="my-3"></div>
      <v-list-item
        color="blue"
        dark
      >List Item</v-list-item>
      <div class="my-3"></div>

      <v-chip-group
        multiple
        column
        color="#e8a741"
      >
        <v-chip
          v-for="(tag, i) in tags"
          :key="tag"
          :text-color="i === 0 ? 'pink' : ''"
        >
          {{ tag }}
        </v-chip>
      </v-chip-group>
    </v-card-text>

    <v-btn-toggle
      v-model="toggle_exclusive"
      color="primary"
    >
      <v-btn>
        <v-icon>format_align_left</v-icon>
      </v-btn>
      <v-btn>
        <v-icon>format_align_center</v-icon>
      </v-btn>
      <v-btn>
        <v-icon>format_align_right</v-icon>
      </v-btn>
      <v-btn>
        <v-icon>format_align_justify</v-icon>
      </v-btn>
    </v-btn-toggle>
    <v-list>
      <v-list-item-group
        v-model="model"
        color="#e8a741"
      >
        <v-list-item
          v-for="(item, i) in items"
          :key="i"
          :color="i === 0 ? 'pink' : ''"
        >
          <v-list-item-icon>
            <v-icon v-text="item.icon"></v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title v-text="item.text"></v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list-item-group>
      <v-list-group color="#e8a741">
        <template v-slot:activator>
          <v-list-item-content>
            <v-list-item-title>Title</v-list-item-title>
          </v-list-item-content>
        </template>
        <v-list-item
          v-for="(item, i) in items"
          :key="i"
          :color="i === 0 ? 'pink' : ''"
          :to="i === 0 ? '/page1' : '/page2'"
        >
          <v-list-item-icon>
            <v-icon v-text="item.icon"></v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title v-text="item.text"></v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list-group>
    </v-list>

    <div class="my-5"></div>

    <v-card
      max-width="400"
      class="mx-auto"
    >
      <v-system-bar></v-system-bar>

      <v-layout
        align-center
        pa-3
      >
        <span class="mr-3">To</span>
        <v-menu
          v-model="menu"
          bottom
          right
          transition="scale-transition"
          origin="top left"
        >
          <template v-slot:activator="{ on }">
            <v-chip
              pill
              v-on="on"
            >
              <v-avatar left>
                <v-img src="https://cdn.vuetifyjs.com/images/john.png"></v-img>
              </v-avatar>
              John Leider
            </v-chip>
          </template>
          <v-card width="300">
            <v-list dark>
              <v-list-item>
                <v-list-item-avatar>
                  <v-img src="https://cdn.vuetifyjs.com/images/john.png"></v-img>
                </v-list-item-avatar>
                <v-list-item-content>
                  <v-list-item-title>John Leider</v-list-item-title>
                  <v-list-item-subtitle>john@vuetifyjs.com</v-list-item-subtitle>
                </v-list-item-content>
                <v-list-item-action>
                  <v-btn
                    icon
                    @click="menu = false"
                  >
                    <v-icon>mdi-close-circle</v-icon>
                  </v-btn>
                </v-list-item-action>
              </v-list-item>
            </v-list>
            <v-list>
              <v-list-item @click="() => {}">
                <v-list-item-action>
                  <v-icon>mdi-briefcase</v-icon>
                </v-list-item-action>
                <v-list-item-subtitle>john@gmail.com</v-list-item-subtitle>
              </v-list-item>
            </v-list>
          </v-card>
        </v-menu>
      </v-layout>

      <v-divider></v-divider>

      <v-text-field
        full-width
        value="Re: Vacation Request"
        label="Subject"
        single-line
      ></v-text-field>

      <v-textarea
        full-width
        single-line
        label="Message"
      ></v-textarea>
    </v-card>

    <div class="my-5"></div>

    <v-bottom-navigation
      v-model="bottomNav"
      color="purple"
      grow
    >
      <v-btn
        text
        color="teal"
        value="recent"
      >
        <span>Recent</span>
        <v-icon>history</v-icon>
      </v-btn>

      <v-btn
        value="favorites"
      >
        <span>Favorites</span>
        <v-icon>favorite</v-icon>
      </v-btn>

      <v-btn
        value="nearby"
      >
        <span>Nearby</span>
        <v-icon>place</v-icon>
      </v-btn>
    </v-bottom-navigation>

    <div class="my-5"></div>

    <v-item-group multiple>
      <v-subheader>Tags</v-subheader>
      <v-item
        v-for="n in 8"
        :key="n"
      >
        <v-chip
          slot-scope="{ active, toggle }"
          active-class="purple--text"
          :input-value="active"
          @click="toggle"
        >
          Tag {{ n }}
        </v-chip>
      </v-item>
    </v-item-group>

    <div class="my-5"></div>

    <v-list
      dark
      nav
    >
      <v-list-item-group
        v-model="model"
        active-class="pink"
      >
        <v-list-item
          v-for="(item, i) in items"
          :key="i"
        >
          <v-list-item-icon>
            <v-icon v-text="item.icon"></v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title v-text="item.text"></v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list-item-group>
    </v-list>

    <div class="my-5"></div>

    <v-card
      class="mx-auto"
      max-width="300"
      tile
    >
      <v-navigation-drawer>
        <v-system-bar></v-system-bar>
        <v-list>
          <v-list-item>
            <v-list-item-avatar>
              <v-img src="https://cdn.vuetifyjs.com/images/john.png"></v-img>
            </v-list-item-avatar>
          </v-list-item>

          <v-list-item link>
            <v-list-item-content>
              <v-list-item-title class="title">John Leider</v-list-item-title>
              <v-list-item-subtitle>john@vuetifyjs.com</v-list-item-subtitle>
            </v-list-item-content>

            <v-list-item-action>
              <v-icon>mdi-menu-down</v-icon>
            </v-list-item-action>
          </v-list-item>
        </v-list>
        <v-divider></v-divider>
        <v-list
          nav
          dense
        >
          <v-list-item-group v-model="item">
            <v-list-item
              v-for="(item, i) in items2"
              :key="i"
            >
              <v-list-item-icon>
                <v-icon v-text="item.icon"></v-icon>
              </v-list-item-icon>

              <v-list-item-content>
                <v-list-item-title v-text="item.text"></v-list-item-title>
              </v-list-item-content>
            </v-list-item>
          </v-list-item-group>
        </v-list>
      </v-navigation-drawer>
    </v-card>
  </v-card>
</template>

<script>
  export default {
    data: () => ({
      bottomNav: 'recent',
      tags: [
        'Work',
        'Home Improvement',
        'Vacation',
        'Food',
        'Drawers',
        'Shopping',
        'Art',
        'Tech',
        'Creative Writing'
      ],
      toggle_exclusive: null,
      item: 0,
      items2: [
        { text: 'My Files', icon: 'mdi-folder' },
        { text: 'Shared with me', icon: 'mdi-account-multiple' },
        { text: 'Starred', icon: 'mdi-star' },
        { text: 'Recent', icon: 'mdi-history' },
        { text: 'Offline', icon: 'mdi-check-circle' },
        { text: 'Uploads', icon: 'mdi-upload' },
        { text: 'Backups', icon: 'mdi-cloud-upload' }
      ],
      items: [
        {
          icon: 'mdi-inbox',
          text: 'Inbox'
        },
        {
          icon: 'mdi-star',
          text: 'Star'
        },
        {
          icon: 'mdi-send',
          text: 'Send'
        },
        {
          icon: 'mdi-email-open',
          text: 'Drafts'
        }
      ],
      model: null,
      menu: false
    })
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/mixins/routable/index.ts [hunk 1]
```diff
@@ -19,33 +23,71 @@ export default Vue.extend({
...
+        this.$listeners.click ||
+        this.$listeners['!click'] ||
+        this.$attrs.tabindex
+      )
+    },
+    isLink (): boolean {
+      return this.to || this.href || this.link
+    },
+    styles (): object {
+      return {}
```
Comment (sh7dm) [inline packages/vuetify/src/mixins/routable/index.ts]: `styles: () => ({})`

File: packages/vuetify/src/mixins/routable/index.ts [hunk 2]
```diff
@@ -19,33 +23,71 @@ export default Vue.extend({
...
       let exact = this.exact
       let tag
 
       const data: VNodeData = {
-        attrs: { disabled: this.disabled },
-        class: classes,
-        style: styles,
+        attrs: {
+          disabled: this.disabled,
+          tabindex: 'tabindex' in this.$attrs ? this.$attrs.tabindex : undefined
```
Comment (sh7dm) [inline packages/vuetify/src/mixins/routable/index.ts]: `...this.$attrs`
or
`tabindex: this.$attrs.tabindex || undefined`
Please explain

File: packages/vuetify/src/mixins/routable/index.ts [hunk 3]
```diff
@@ -19,33 +23,71 @@ export default Vue.extend({
...
   computed: {
+    classes (): object {
+      return {
+        [this.activeClass]: this.isActive,
+        [this.proxyClass]: this.isActive
+      }
+    },
     computedRipple (): RippleOptions | boolean {
-      return (this.ripple && !this.disabled) ? this.ripple : false
+      return this.ripple != null ? this.ripple : !this.disabled && this.isClickable
```
Comment (sh7dm) [inline packages/vuetify/src/mixins/routable/index.ts]:
```suggestion
      return this.ripple || !this.disabled && this.isClickable
```

File: packages/vuetify/src/mixins/routable/index.ts [hunk 4]
```diff
@@ -19,33 +23,71 @@ export default Vue.extend({
...
+    link: Boolean,
     href: [String, Object],
     to: [String, Object],
     nuxt: Boolean,
     replace: Boolean,
-    ripple: [Boolean, Object],
+    ripple: {
+      type: [Boolean, Object],
+      default: null
+    },
```
Comment (sh7dm) [inline packages/vuetify/src/mixins/routable/index.ts]:
```suggestion
    } as PropValidator<null | boolean | object>,
```

File: packages/vuetify/src/mixins/routable/index.ts [hunk 5]
```diff
@@ -1,8 +1,12 @@
 import Vue, { VNodeData } from 'vue'
 import { PropValidator } from 'vue/types/options'
```
Comment (sh7dm) [inline packages/vuetify/src/mixins/routable/index.ts]:
```suggestion

// Types
import { PropValidator } from 'vue/types/options'
```

File: packages/vuetify/src/mixins/routable/index.ts (hunk 2 above)
Comment (MajesticPotatoe) [reply packages/vuetify/src/mixins/routable/index.ts]: in the first scenario, you only want `tabindex` from `this.$attrs` not everything from it
in the second scenario, `this.$attrs.tabindex` will throw an error if `this` or `this.$attrs` is undefined
🤷‍♂

File: packages/vuetify/src/mixins/routable/index.ts (hunk 3 above)
Comment (KaelWD) [reply packages/vuetify/src/mixins/routable/index.ts]: ripple can be false

File: packages/vuetify/src/mixins/routable/index.ts [hunk 6]
```diff
@@ -92,6 +134,20 @@ export default Vue.extend({
       if (this.target) data.attrs!.target = this.target
 
       return { tag, data }
-    }
+    },
+    onRouteChange () {
+      if (!this.to || !this.$refs.link) return
+      const activeClass = `${this.activeClass} ${this.proxyClass || ''}`.trim()
+
+      const path = `_vnode.data.class.${activeClass}`
```
Comment (KaelWD) [inline packages/vuetify/src/mixins/routable/index.ts]: https://github.com/vuejs/vue-router/blob/8ac478f4f6da314e7bcbdbee8372cc18b1108517/src/util/route.js#L73-L94

File: packages/vuetify/src/components/VList/VListGroup.ts [hunk 7]
```diff
@@ -129,7 +132,10 @@ export default baseMixins.extend<options>().extend({
       return this.$createElement(VListItem, {
         staticClass: 'v-list-group__header',
         class: {
-          [this.activeClass]: this.isActive
+          [this.activeClass || '']: this.isActive
```
Comment (DRoet) [inline packages/vuetify/src/components/VList/VListGroup.ts]: nit: why not default the prop to an empty string?

File: packages/vuetify/src/mixins/routable/index.ts (hunk 4 above)
Comment (johnleider) [reply packages/vuetify/src/mixins/routable/index.ts]: Is this necessarily needed? We do not do this anywhere else that we use Ripple

**Pull request merged**